src/bloom.cpp

```diff
@@ -242,9 +264,20 @@ bool CRollingBloomFilter::contains(const std::vector<unsigned char>& vKey) const
     return b1.contains(vKey);
 }
 
-void CRollingBloomFilter::clear()
+bool CRollingBloomFilter::contains(const uint256& hash) const
 {
-    b1.clear();
-    b2.clear();
+    if (nInsertions < nBloomSize / 2) {
+        return b2.contains(hash);
+    }
+    return b1.contains(hash);
+}
+
+void CRollingBloomFilter::reset(unsigned int nNewTweak)
+{
+    if (!nNewTweak)
+        nNewTweak = GetRand(std::numeric_limits<unsigned int>::max());
```

Comment:
Perhaps, but that sounds like a fundemental API flaw in GetRand() - it should fail hard under that circumstance. 

According to https://www.openssl.org/docs/crypto/RAND_bytes.html "An error occurs if the PRNG has not been seeded with enough randomness to ensure an unpredictable byte sequence.", and we do check that the underlying RAND_bytes() returns success. If we trust OpenSSL, then the fact that Bitcoin Core isn't failing at startup with this code implies everything is working and the filter is getting initialized with genuine randomness.

Of course, that may be all a pile of bullshit... but lets see if we can fix the underlying issue first.